crates/cli-support/src/js/mod.rs

```diff
@@ -640,35 +640,43 @@ impl<'a> Context<'a> {
             }
         }
 
-        let default_module_path = if !self.config.omit_default_module_path {
-            match self.config.mode {
-                OutputMode::Web => format!(
-                    "\
-                    if (typeof input === 'undefined') {{
-                        input = new URL('{stem}_bg.wasm', import.meta.url);
-                    }}",
-                    stem = self.config.stem()?
-                ),
-                OutputMode::NoModules { .. } => "\
-                    if (typeof input === 'undefined') {
-                        let src;
-                        if (typeof document === 'undefined') {
-                            src = location.href;
-                        } else {
-                            src = document.currentScript.src;
-                        }
-                        input = src.replace(/\\.js$/, '_bg.wasm');
-                    }"
-                .to_string(),
-                _ => "".to_string(),
+        let (default_module_path_creator, default_input_initializer): (String, &'static str) = {
+            const ASSIGNMENT_STRING: &'static str = " = default_wasm_source_url";
+            const EMPTY: (String, &'static str) = (String::new(), "");
+            let config = &self.config;
+
+            if !config.omit_default_module_path {
+                match config.mode {
+                    OutputMode::Web => (
+                        format!(
+                            "const default_wasm_source_url = new URL('{stem}_bg.wasm', import.meta.url);",
+                            stem = config.stem()?
+                        ),
+                        ASSIGNMENT_STRING,
+                    ),
+                    OutputMode::NoModules { .. } => (
+                        r"\
+                            // Document#currentScript is a getter,
+                            // it's value changes upon evaluating each script,
+                            // therefore it must be cached
+                            const default_wasm_source_url = (typeof document !== 'undefined'
+                                ? document.currentScript.src
+                                : location.href
+                            ).replace(/\.js$/, '_bg.wasm');
+                        "
+                        .to_string(),
+                        ASSIGNMENT_STRING,
+                    ),
+                    _ => EMPTY,
+                }
+            } else {
+                EMPTY
             }
-        } else {
-            String::from("")
         };
 
         let ts = self.ts_for_init_fn(
             has_memory,
-            !self.config.omit_default_module_path && !default_module_path.is_empty(),
+            !self.config.omit_default_module_path && !default_module_path_creator.is_empty(),
```

Comment:
If I'm following this logic correctly, the entire expression could be shortened to the following:
```rust
!default_module_path_creator.is_empty()
```

cc @diceride

If someone omits the default path, and chooses, say, Node.js, there still won't be a default path, therefore we can just check if there is a default path via `.is_empty()`.